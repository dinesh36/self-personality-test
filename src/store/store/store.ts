import {
  createStore, applyMiddleware, combineReducers, AnyAction, EmptyObject, Store,
} from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import layoutReducer from './layout/layout.reducer';
import { IRootStore } from './store.type';
import testReducer from './test/test.reducer';

let cachedStore: (
  Store<EmptyObject & IRootStore, AnyAction> & { dispatch: unknown; }) | null = null;

export const getStaticReducer = () => ({
  layoutState: layoutReducer,
  testState: testReducer,
});

export const configureStore = () => {
  cachedStore = createStore(
    combineReducers(getStaticReducer()),
    composeWithDevTools(applyMiddleware(thunk)),
  );
  return cachedStore;
};

export const getStore = () => {
  if (cachedStore !== null) {
    return cachedStore;
  }
  return configureStore();
};
