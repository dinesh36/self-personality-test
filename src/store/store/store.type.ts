export interface IRootStore{
  layoutState: {
    title: string
  }
}
