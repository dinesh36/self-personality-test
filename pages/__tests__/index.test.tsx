import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Home from '../index';
import { getStore } from '../../src/store/store/store';

it('changes the class when hovered', () => {
  const store = getStore();
  const component = renderer.create(
    <Provider store={store}>
      <Home />
    </Provider>,
  );
  const cmp = component.toJSON();
  expect(cmp).toMatchSnapshot();
});
