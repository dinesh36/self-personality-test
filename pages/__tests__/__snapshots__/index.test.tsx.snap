// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the home page 1`] = `
<div
  className="home"
>
  <img
    alt="logo"
    className="homeImage"
    src="images/app-logo.png"
  />
  <div
    className="card "
  >
    <div
      className="subTitle"
    >
      <img
        alt="note"
        src="images/home/note.svg"
      />
      Are you an introvert or an extrovert?
    </div>
    <p>
      These two personality types are opposites â€” introverts focus inward, into their own thoughts, and extroverts focus outward, into the world. But don't get too excited; most people are a little bit of both. Do you want to know what kind of person you are ? Let's Check
    </p>
    <a
      href="/test"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <button
        className="ant-btn css-dev-only-do-not-override-1n7nwfa ant-btn-primary button startButton"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span>
          Start Test
        </span>
      </button>
    </a>
  </div>
</div>
`;
